src/components/plots/MitochondrialContentPlot.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { Spin } from 'antd';
+import { Vega } from 'react-vega';
+
+import PlatformError from '../PlatformError';
+import { generateSpec, generateData } from '../../utils/plotSpecs/generateMitochondrialContentSpec';
+import { loadEmbedding } from '../../redux/actions/embedding';
+import loadCellMeta from '../../redux/actions/cellMeta';
+import { loadCellSets } from '../../redux/actions/cellSets';
+import { loadProcessingSettings } from '../../redux/actions/experimentSettings';
+
+const MitochondrialContentPlot = (props) => {
+  const { experimentId, config } = props;
+  const defaultEmbeddingType = 'umap';
+  const plotName = 'mitochondrialContent';
+
+  const dispatch = useDispatch();
+
+  const embeddingSettings = useSelector(
+    (state) => state.experimentSettings.processing?.configureEmbedding?.embeddingSettings,
+  );
+  const embedding = useSelector((state) => state.embeddings[embeddingSettings.method]) || {};
+
+  const mitochondrialContent = useSelector((state) => state.cellMeta?.mitochondrialContent) || {};
+  const cellSets = useSelector((state) => state.cellSets);
+  const [plotSpec, setPlotSpec] = useState({});
+
+  useEffect(() => {
+    if (cellSets.loading && !cellSets.error) {
+      dispatch(loadCellSets(experimentId));
```

Comment:
This is more to discuss, so let me know what you think about it.

This is something I noticed differs with other implementations of plots. In these cases, the plots are making the dispatches themselves while also managing the rendering directly, which means they refresh themselves on their own, while for example the elbow plot is just setting up the spec and receiving the data to display.

I feel like this option results in more repeated code since, for example, the `cellSets` loading part is always the same while perhaps by having the plot receive them as props we could avoid having it in all the embedding plots code.

And something similar goes for the processing settings, loadEmbedding (which differs only in the method) and others.

Even if this was a good idea it would probably take a lot of refactoring, so if you think this is worth trying then we could add a ticket for it or perhaps implement something less refactor-y for now like just separating the dispatches and useSelector/useState's by putting them in a wrapper for the empty embedding plot component.

Reply:
Something nice about this option too could be avoiding having one embedding for the each type of data it displays, instead having one loader for each type of data and only one single embedding plot generator

Reply:
Yea, it would be good to be able to reuse plots with similar functionality. Generating the plot is also related with the plot's specGenerator and dataGenerator.  There may be a way to better implement these separate functions if we want to refactor the plot element.

I believe this can be put out as a ticket, as it requires some investigation how to best refactor the components.

Reply:
On separate loading, it's good to move the dispatches outside the plots. However, this means any data loading the plot requires should be done in the data i.e. parent has to know the dependencies in child data. This can be alleviated with a good interface from the child, so it forces the parent to provide the required data. I think this goes hand in hand with refactoring of the plot components.

Reply:
So yea I think we can move this refactoring out for another ticket.